app/models/step.rb

```diff
@@ -33,6 +34,10 @@ def pre_order_tree_traversal
     [self] + self.child_approvals.flat_map(&:pre_order_tree_traversal)
   end
 
+  def completer
```

Comment:
oh interesting, since we already have a `user` relation, what about just adding a `completed_at` timestamp to `step` and setting that when it is completed? Seems redundant to have 2 foreign keys going to the same person, as far as I can tell...